### DISCUSSION TITLE
トラックボールが反応しません

### DISCUSSION THREAD

**Discussion by 8oceanq** [Yowkees/keyball#595] [Q&A]
keyball61を組み立てましたが、トラックボールの反応がありません。
はんだ付け等問題点がありましたらご教示いただけますと幸いです。
写真の問題点等ありましたらご指摘ください。

![20240530_090420](https://github.com/Yowkees/keyball/assets/171204510/f01e3edd-b51f-4533-b0b7-29436f738cc0)
![Uploading 20240530_090503.jpg…](
![20240530_090425](https://github.com/Yowkees/keyball/assets/171204510/9ee54f7f-fcdf-48fc-854c-6fb899492a2c)
![20240530_090430](https://github.com/Yowkees/keyball/assets/171204510/0470bbd7-38b1-4d1a-8d53-dc794ce5830e)
![20240530_090442](https://github.com/Yowkees/keyball/assets/171204510/f5a919b5-be24-49a9-ad64-4162e8476334)
![20240530_090452](https://github.com/Yowkees/keyball/assets/171204510/1d792312-72fd-4e04-9cb9-66bfe9be60d3)
![20240530_090501](https://github.com/Yowkees/keyball/assets/171204510/42b544b1-769e-4d11-8725-7ae3f1f62efe)

![20240530_085922](https://github.com/Yowkees/keyball/assets/171204510/1c185927-ea26-4558-8513-4bca2aea6026)

**Comment by Ryo624** [comment]
トラックボール基板はレンズではんだ付けが見えづらいですがL字コンスルーやProMicro裏と表のはんだが多すぎるように見えます。
特にProMicro裏側ははんだが隣の端子に触れているとショートしてしまい不具合につながるので、全体的にはんだの量を少し減らしてはんだの形が富士山のように山型になるように意識して端子に馴染むようにはんだ付けを行ってください。

**Reply by 8oceanq** [reply]
ご回答ありがとうございます。
ご指摘に基づき、トラックボール基盤も含めてはんだを減らしてみましたが、状況は変わらず、トラックボールの反応がありません。
はんだを減らした後の画像も添付していますので、はんだが多いなどありましたらご指摘お願いいたします。
![20240530_185418](https://github.com/Yowkees/keyball/assets/171204510/069128a5-96de-4bee-be36-022f08adba81)
![20240530_185511](https://github.com/Yowkees/keyball/assets/171204510/f2e57c5a-8c46-4f95-a8ac-f434bfb09de1)
![20240530_185517](https://github.com/Yowkees/keyball/assets/171204510/08c6b976-ba05-480b-95f3-34a1b581aac3)
![20240530_184201](https://github.com/Yowkees/keyball/assets/171204510/4acb46e3-065c-45f3-a573-d0b17bfb8025)

**Comment by Ryo624** [comment]
はんだの量は問題ないかと思います。
添付させていただいたProMicroのはんだが上手く端子に馴染んでいないように見えます。
トラックボール基板のはんだ付けももう少し山型で端子に馴染むようにはんだをしてみてください。

![335136075-069128a5-96de-4bee-be36-022f08adba81](https://github.com/Yowkees/keyball/assets/143698349/e36eb64d-f4be-4286-827e-7d97450e0d4a)

**Reply by 8oceanq** [reply]
ご指摘に従い、はんだを修正してみましたが、やはり反応がありません。
remapを見てみましたが、トラックボールが認識されていない、両方親指キーが３つの状態で認識されています。
Promicroの写真とトラックボール部分を装着した写真を添付していますので、修正点ありましたらご指摘いただければ幸いです。
![20240531_101214](https://github.com/Yowkees/keyball/assets/171204510/57795fff-6afb-4569-a3f1-470c4268808e)
![20240531_101225](https://github.com/Yowkees/keyball/assets/171204510/fe6c5c6c-804d-4c65-864a-ccbbf6c317f8)
![20240531_101243](https://github.com/Yowkees/keyball/assets/171204510/3a30306f-8633-4383-a3d3-cd66a8738e01)
![20240531_101258](https://github.com/Yowkees/keyball/assets/171204510/123e0550-b721-47fd-9e15-d96008f88409)

**Comment by Ryo624** [comment]
ファームウェアの書き込みのやり直しを行っても改善しないでしょうか。
ProMicroの左右も入れ替えてみてください。

**Reply by 8oceanq** [reply]
ご連絡ありがとうございます。

ファームウェアの書き込みについては改善しませんでした。
Promicroの左右入れ替えの際にピンソケットを破損してしまいましたので、用意できましたら試してみます。
またご質問差し上げるかもしれませんがその際はよろしくお願いします。

**Reply by 8oceanq** [reply]
> ファームウェアの書き込みのやり直しを行っても改善しないでしょうか。 ProMicroの左右も入れ替えてみてください。

Promicroとピンソケットの交換を行いましたが改善しませんでした。
他の方法がございましたらご教示いただけますと幸いです。

また、Promicroとピンソケットを交換後、親指キーの外側とPromicroの真下のキーが反応しなくなってしまいました。
気になる点がございましたらご指摘いただければ幸いです。

![20240603_124356](https://github.com/Yowkees/keyball/assets/171204510/ccfaa55e-69f6-4743-b2f9-7398917142a2)
![20240603_124407](https://github.com/Yowkees/keyball/assets/171204510/0ecbd8d7-20a8-4e7c-9553-c2d63f35f063)
![20240603_124422](https://github.com/Yowkees/keyball/assets/171204510/8b9d6d6e-bc53-41a0-8248-a86c8b76061d)
![20240603_124435](https://github.com/Yowkees/keyball/assets/171204510/2a40a880-9bec-4b73-ab30-ad970f1a1bf4)
![20240603_124504](https://github.com/Yowkees/keyball/assets/171204510/8ed5c651-07c9-42c6-aac1-e47192ec1ca6)
